### PR TITLE
TF-657 Send a read receipt when asked by the sender

### PR DESCRIPTION
**Pull request by dab246**
#657 

### Resolved

https://user-images.githubusercontent.com/80730648/196721710-4ee4be62-f13e-45cf-a44b-58b38474ff9c.mov

https://user-images.githubusercontent.com/80730648/196722499-c9aced11-cd2a-4293-9f05-eca7d1b584fd.mov

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -246,4 +246,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 91efd4097cc41f032416d67ecaac21cc2de8dd55
 
-COCOAPODS: 1.11.3
+COCOAPODS: 1.11.2
```
Comment (hoangdat) [inline ios/Podfile.lock:249]: Why we need to downgrade this?

File: lib/features/base/interactors_bindings.dart [hunk 2]
```diff
@@ -0,0 +1,24 @@
+
+import 'package:get/get_instance/src/bindings_interface.dart';
+
+abstract class InteractorsBindings extends Bindings {
```
Comment (hoangdat) [inline lib/features/base/interactors_bindings.dart:4]: Why we need this class?

File: lib/features/email/data/network/email_api.dart [hunk 3]
```diff
@@ -27,17 +27,7 @@ import 'package:jmap_dart_client/jmap/mail/email/submission/envelope.dart';
...
-import 'package:model/download/download_task_id.dart';
-import 'package:model/email/attachment.dart';
-import 'package:model/email/mark_star_action.dart';
-import 'package:model/email/read_actions.dart';
-import 'package:model/extensions/email_extension.dart';
-import 'package:model/extensions/keyword_identifier_extension.dart';
-import 'package:model/extensions/list_email_id_extension.dart';
-import 'package:model/extensions/mailbox_id_extension.dart';
-import 'package:model/oidc/token.dart';
+import 'package:model/model.dart';
```
Comment (hoangdat) [inline lib/features/email/data/network/email_api.dart:30]: Not a good practice

File: lib/features/base/interactors_bindings.dart (hunk 2 above)
Comment (dab246) [reply lib/features/base/interactors_bindings.dart:4]: I want to separate the injecting of classes that handle the data and domain layers

File: ios/Podfile.lock (hunk 1 above)
Comment (dab246) [reply ios/Podfile.lock:249]: Oh. It is spawned automatically when run on iOS 16. It's not necessary. I will remove it.

**Comment by dab246** [comment]
> 



> Let's improve the massage content...
> 
> Screenshot from 2022-10-20 08-48-38
> 
> -> Line breaks.
> 
> ```
> Message was read by Qa Test 1 <...> i=on Wed. 19OCT
> 
> Subject: YYY
> 
> Note: This Return Receipt only acknowledges that the message was displayed on the recipient's computer. There is no guarantee that the recipient has read or understood the message contents.
> ```

@chibenwa 

Does the content `send` have the correct form?
```
Message was read by Qa Test 1 <...> i=on Wed. 19OCT

Subject: YYY

Note: This Return Receipt only acknowledges that the message was displayed on the recipient's computer. There is no guarantee that the recipient has read or understood the message contents.
```

**Comment by chibenwa** [comment]
Yes, textual content is free form text.

File: core/lib/presentation/views/dialog/confirmation_dialog_builder.dart [hunk 4]
```diff
@@ -109,27 +123,38 @@ class ConfirmDialogBuilder {
...
-              alignment: Alignment.centerRight,
-              child: Padding(
-                padding: EdgeInsets.only(top: 8, right: 8),
-                child: buildIconWeb(
-                    icon: SvgPicture.asset(_imagePath.icCloseMailbox, fit: BoxFit.fill),
-                    onTap: () => _onCloseButtonAction?.call())
-              )),
+                alignment: Alignment.centerRight,
+                child: Padding(
+                    padding: EdgeInsets.only(top: 8, right: 8),
```
Comment (sherlockvn) [inline core/lib/presentation/views/dialog/confirmation_dialog_builder.dart:153]: can use const here

File: core/lib/presentation/views/dialog/confirmation_dialog_builder.dart [hunk 5]
```diff
@@ -109,27 +123,38 @@ class ConfirmDialogBuilder {
...
-        padding: EdgeInsets.zero,
-        width: 400,
+    if (showAsBottomSheet) {
+      return _bodyContent();
+    } else {
+      return Dialog(
+        key: _key,
+        shape: RoundedRectangleBorder(
+            borderRadius: BorderRadius.all(Radius.circular(16))),
+        insetPadding: EdgeInsets.symmetric(
```
Comment (sherlockvn) [inline core/lib/presentation/views/dialog/confirmation_dialog_builder.dart:133]: can use const here

**Review by ManhNTX — APPROVED** [review]
LGTM

File: core/lib/presentation/views/dialog/confirmation_dialog_builder.dart (hunk 4 above)
Comment (dab246) [reply core/lib/presentation/views/dialog/confirmation_dialog_builder.dart:153]: @sherlockvn Can you write a sample for me?

**Pull request merged**